### ISSUE TITLE
Improve the labeling of EntitiesSavedStates when used with modal behavior

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#67354]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/issues/67313 / https://github.com/WordPress/gutenberg/pull/67351

When it is used with a modal behavior attached, the EntitiesSavedStates component has a `role=dialog' and it's labelled via an `aria-labelledby` attribute that points to these part:

https://github.com/WordPress/gutenberg/blob/f8140c4fcc8db2d6078ad76fd433c79df3543860/packages/editor/src/components/entities-saved-states/index.js#L163-L171

The problem with that, is that when `additionalPrompt` is passed, it makes part of the dialog labeling which:
- Makes the dialog labeling potentially too long and misleading.
- The `additionalPrompt` text is meant to be more an additional _description_ rather than a label.

For example. in the Theme preview in the Site editor the label of the dialog is as follows:

`Are you ready to save? Saving your changes will change your active theme from Twenty Twenty-Five to Twenty Twenty-Four."`

That's used for the _label_ and it's way too long. The second part should be used for the description and the labeling should only be the first part: `Are you ready to save?`.

![Image](https://github.com/user-attachments/assets/fd7815da-c7e2-4d8d-900a-bb79d821016d)

### Step-by-step reproduction instructions

First, apply the changes from https://github.com/WordPress/gutenberg/pull/67351 if it's hasn't been merged yet as that PR fixes the labeling of the dialog.
- Go to the WP admin > Appearance > any non-acitve block theme > Live preview.
- The editor opens to preview the theme.
- Click the editor canvas to switch to edit mode.
- Click the button 'Activate {theme name}' at the top right of the screen.
- Inspect the DOM and select the element with role=dialog and class `entities-saved-states__panel`.
- In the dev tools inspector accessibility tab observe that the text passed as `additionalPrompt` is used in the labeling of the element.

![Image](https://github.com/user-attachments/assets/b1e286ae-d5cf-4a55-9589-769307a8dedc)

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
